iceoryx2-cal/tests/arc_sync_policy_trait_tests.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#[generic_tests::define]
+mod arc_sync_policy {
```

Comment:
Can you add some `compile_fail` tests to ensure the `SingleThreaded` policy does not break?